### PR TITLE
Run tsc in PRs and regular builds

### PR DESCRIPTION
**Pull request by forabi**
This should catch some errors before attempting to deploy

### CODE REVIEW

File: deploy.js [hunk 1]
```diff
@@ -20,25 +28,32 @@ const secrets = [
...
+  commands = ['yarn test'];
+
   if (BRANCH !== 'master') {
-    console.info('Not on master branch. Nothing to do.');
-    process.exit(0);
+    console.info('Skipping deploying on non-master branches');
+  } else if (isPullRequest === true) {
+    console.info('Skipping deployment commands in PRs');
+  } else {
+    commands = [
```
Comment (wholesomedev) [inline deploy.js]: This is fine, but you can also do

```js
commands.push(
  item1,
  item2,
  item3,
)
```

**Review by wholesomedev — COMMENTED** [review]
What about linting, should also lint PRs in the CI?

File: deploy.js (hunk 1 above)
Comment (forabi) [reply deploy.js]: I've been writing JS for years, and I didn't know that `array.push` accepts multiple arguments 😅 
Anyway, I still prefer the spread syntax.

**Review by wholesomedev — DISMISSED** [review]

**Comment by forabi** [comment]
@msafih I've just added `eslint` and `validate-files-names` to test command. I think we should add these to the other repos too.

**Review by wholesomedev — DISMISSED** [review]

**Review by wholesomedev — DISMISSED** [review]

**Review by wholesomedev — APPROVED** [review]

**Pull request merged**